crates/astria-sequencer/src/grpc/sequencer.rs

```diff
@@ -125,9 +125,13 @@ impl SequencerService for SequencerServer {
             .into_iter()
             .map(RollupId::to_vec)
             .collect::<Vec<_>>();
+        let rollup_ids_with_data: Vec<RollupId> = rollup_ids
+            .into_iter()
+            .filter(|id| all_rollup_ids.contains(&id.to_vec()))
```

Comment:
This is pretty expensive because with `O(n)` behaviour for every single check. Also, unnecessary heap allocations everywhere. I would suggest doing the following instead:
```rust
let all_rollup_ids = snapshot
            .get_rollup_ids_by_block_hash(&block_hash)
            .await
            .map_err(|e| Status::internal(format!("failed to get rollup ids from storage: {e}")))?
            .into_iter()
            .collect::<HashSet<_>>();
for rollup_id in rollup_ids {
    if all_rollup_ids.contains(&rollup_id) {
        // perform the lookup, push to `rollup_transactions`
    }
}
let all_rollup_transactions = all_rollup_transactions.into_iter().map(RollupId::to_vec).collect();
```